### PR TITLE
Time window edit

### PR DESCRIPTION
**Pull request by gpinkley**
Ticket_1349
Ticket_1347

Reworked the TimeFilter: The TimeFilter was changed to have four possible settings:

Center Time and Window: This is a setting where a center time is specified and a time window defined around the center time.
Start Time and Window: This is a setting where a start time is specified, and a window of time after that start time is defined.
End Time and Window: This is a setting where an end time is specified, and a window of time before that end time is defined.
Start Time and End Time: This is a setting where both a start time and an end time are specified, and the window of time is the interval between them.

Added setting to choose dateTimeSettings under Settings>General Settings: A new setting was added in the "General Settings" that allows users to choose between the different TimeFilter settings.

Made the hide filters EventSearchNavbar reflect the corresponding start/end time/window: The EventSearchNavbar component was updated to show the appropriate heading based on the selected TimeFilter setting.

### CODE REVIEW

**Comment by elwills** [comment]
Looks like when you hide the filter with `Start` or `End` only, it shows the +/- window instead of the total.

**Comment by gpinkley** [comment]
> Looks like when you hide the filter with `Start` or `End` only, it shows the +/- window instead of the total.

It is currently showing Start: Tue Mar 16 2021 12:58:57 GMT-0400 + a day. Am I understand. you want it to show the start and end date there?

**Comment by elwills** [comment]
image
image

File: SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchSlice.ts [hunk 1]
```diff
@@ -157,7 +163,12 @@ export const EventSearchsSlice = createSlice({
             sagType: 'both', swellType: 'both', transientType: 'both',
             curveID: 1, curveInside: true, curveOutside: true
         },
-        TimeRange: { date: moment.utc().format(momentDateFormat), time: '12:00:00.000', windowSize: 7, timeWindowUnits: 4 },
+        TimeRange: {
+            date: moment.utc().subtract(84,'h').format(momentDateFormat),
+            time: '12:00:00.000',
+            windowSize: 84,
+            timeWindowUnits: 5
```
Comment (elwills) [inline SEBrowser/Scripts/TSX/Components/EventSearch/EventSearchSlice.ts:170]: This is weeks. It needs to be `3` for hours.

**Review by elwills — APPROVED** [review]

**Pull request merged**